omod/src/main/java/org/openmrs/module/reportingrest/web/resource/ReportRequestResource.java

```diff
@@ -65,21 +72,80 @@ public ReportRequest newDelegate() {
 	public ReportRequest getByUniqueId(String uuid) {
 		return getService().getReportRequestByUuid(uuid);
 	}
-	
+
 	@Override
 	protected PageableResult doSearch(RequestContext context) {
+		String statusesGroup = context.getParameter("statusesGroup");

```

Comment:
I don't love this hard-coding like this with these statusGroups.  Why is this needed?

Reply:
It adds groups to `ReportRequest.Status`:
- ran - all statuses which mean a report was executed
- processing - all statuses which mean report is being executed
- scheduled - all statuses which mean it's an execution according to schedule

We could change the `statusesGroup` parameter to `status` and allow comma-separated list here, moving the grouping of Status to the FE.

Reply:
@pwargulak - I think that would be preferable.  Basically, allow the endpoint to accept multiple values for `status` and interpret this as an `OR` condition.  I wouldn't accept a csv input, rather I would just get the array of parameter values out of the request for the `status` parameter name, and treat this as an "or".